### PR TITLE
build rest api user

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine
https://edu-redmine.sun-asterisk.vn/issues/75103
## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Đăng ký: 
image
Đăng nhập: 
image
Danh sách user:
image
Chi tiết user:
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/api/admin.user.api.controller.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export const getListUser = async (req: Request, res: Response) => {
+  try {
+    await checkAdmin(req, res);
+    const page = parseInt(req.query.page as string) || DEFAULT_PAGE;
+
+    const { listUser, totalPages } = await getListUserForManage(page);
+
+    res.status(200).json({ listUser, totalPages });
+  } catch (err) {
+    return res.status(500).json({ message: 'Server error' });
```
Comment (huongnt-2545) [inline src/api/admin.user.api.controller.ts:18]: `res.status(500).json({ message: 'Server error' });` -> viết method để tái sử dụng

File: src/api/admin.user.api.controller.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+} from '../service/admin.user.service';
+
+export const getListUser = async (req: Request, res: Response) => {
+  try {
+    await checkAdmin(req, res);
+    const page = parseInt(req.query.page as string) || DEFAULT_PAGE;
+
+    const { listUser, totalPages } = await getListUserForManage(page);
+
+    res.status(200).json({ listUser, totalPages });
```
Comment (huongnt-2545) [inline src/api/admin.user.api.controller.ts:16]: `es.status(200).json({ listUser, totalPages });` -> viết method chung để tái sử dụng e nha

File: src/api/auth.api.controller.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      return res.status(404).json({ message: req.t('login.notfound') });
+    }
+
+    const isMatch = await checkPassword(password, user.password);
+    if (!isMatch) {
+      return res.status(401).json({ message: req.t('login.error') });
+    }
+
+    if (user.disable === true) {
+      return res.status(403).json({ message: 'Account is disabled' });
```
Comment (huongnt-2545) [inline src/api/auth.api.controller.ts:59]: I18n các message e nhé

**Pull request closed without merging**